### PR TITLE
Added interface definition for ht_ndel()

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Relates #108

### CODE REVIEW

File: src/libreset/ht.h [hunk 1]
```diff
@@ -118,6 +120,20 @@ ht_del(
...
+ *
+ * @memberof ht
+ *
+ * @return the number of removed elements.
+ */
+unsigned int
+ht_ndel(
+    struct ht* ht, //!< The hashtable object to delete from
+    void* cmp,
+    r_set_cfg* cfg
```
Comment (neithernut) [inline src/libreset/ht.h]: missing `struct` keyword.

Comment (matthiasbeyer) [inline src/libreset/ht.h]: embarrasing.

File: src/libreset/ht.h [hunk 2]
```diff
@@ -118,6 +120,20 @@ ht_del(
...
+ * Deleting n elements from the hashtable by predicate
+ *
+ * @memberof ht
+ *
+ * @return the number of removed elements.
+ */
+unsigned int
+ht_ndel(
+    struct ht* ht, //!< The hashtable object to delete from
+    void* cmp, //!< element to compare against
```
Comment (neithernut) [inline src/libreset/ht.h]: Wait... for this you would, again, have a prediacate function (with additional argument for user defined data)

Comment (matthiasbeyer) [inline src/libreset/ht.h]: Sure, somehow you have to select the items you want to remove...

**Comment by neithernut** [comment]
ACK.

**Pull request merged**